CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+    and all of it's dependencies is with the following commands:
+    ```
+    git clone https://github.com/cctbx/cctbx_project modules/cctbx_project
+    ln -s modules/cctbx_project/libtbx/auto_build/bootstrap.py
+    python bootstrap.py hot base update build --builder=dials --use-conda
+    ```
+2.  Activate the environment with `source <root>/build/setpaths.sh`. This will
+    need to be done every time you work on dials code.
+3.  Install pre-commit hooks with `libtbx.precommit install`. This will install
+    hooks for Dials, dxtbx and xia2.
+
+The Dials repository is now checked out in `<root>/modules/dials`. During
+development, run tests with pytest with `libtbx.pytest --regression` to ensure
+that all of the tests still pass.
+
+If you update or change dependencies, you may occasionally need to regenerate
+the static libtbx ecosystem by running `libtbx.refresh` - this will also
+generate an alias to any executable scripts installed by `pip` as
+`libtbx.<script>`. Using these helps ensure that you are using the bundled
+python version instead of your system version.
+
+## Essential Code Development Standards
+
+- **Please install the pre-commit hooks**. (`libtbx.precommit install` if using
+  the libtbx ecosystem). These use the [pre-commit] library and ensures that
+  various sanity checks are run before commit, including formatting, syntax
+  compatibility, basic flake8 checks, lack of conflict markers and file size
+  limits. Basically, most of the essential rules will be checked automatically
+  by this.
+- **We format python code with [black]**. This means that while writing code
+  you don't have to worry about laying things out neatly, because black will
+  take care of the formatting. You should try to commit code formatted with
+  black (the pre-commit hook will help do this for you), but if for some reason
+  you can't, the whole codebase is auto-cleaned once a week. Most IDE's and
+  editors have support for running formatters like black frequently or
+  automatically.
+- **Avoid introducing new pre-commit flake8 warnings** - if you feel that it's
+  appropriate to violate a warning, mark it up explicitly with a [noqa]
+  comment. Probably the most common cause of this are "F401 - module imported
+  or unused", which happens when importing packages to collect into a single
+  namespace for other imports (though declaring `__all__` avoids this issue).
+  The pre-commit hooks will pick up the most important of these, but please try
+  to resolve any other valid warnings shown with a normal run of flake8. The
+  configuration in the repository turns off any that disagree with black's
+  interpretation of the rules or standard practice in our repositories.
+- **Code should be written with python 3 syntax** - This means "new" exception
+  syntax, print functions, new-style classes, and a smattering of other changes
+  — the pre-commit hooks check for this.
```

Comment:
I don't think so.

Reply:
I thought we used to have explicit AST checks, and assumed that the lack of "python3" interpreter explicitly was a reflection of the fact that all our hooks are run on python 3 now?

Also, black fails for most of these anyway, I think...?

Reply:
pre-commit black runs on Python 3 and does assume correct syntax. And we do run check-ast, which may be stricter. But that still doesn't check for new-style classes or right-shifted print functions.

Reply:
Ah, "The pre-commit hooks check for this" was referring to " **Code should be written with python 3 syntax**" - technically right shifted print functions are python 3 syntax :)